python/ray/serve/common.py

```diff
@@ -24,6 +28,51 @@ class EndpointInfo:
     route: str
 
 
+class ApplicationStatus(str, Enum):
+    DEPLOYING = "DEPLOYING"
+    RUNNING = "RUNNING"
+    DEPLOY_FAILED = "DEPLOY_FAILED"
+
+
+@dataclass
+class ApplicationStatusInfo:
+    status: ApplicationStatus
+    message: str = ""
+    deployment_timestamp: float = 0
+
+    def __eq__(self, other):
+        if not isinstance(other, ApplicationStatusInfo):
+            return False
+
+        return (
+            self.status == other.status
+            and self.message == other.message
+            and self.deployment_timestamp == other.deployment_timestamp
+        )
```

Comment:
dataclass should already generate this for you: https://docs.python.org/3/library/dataclasses.html#dataclasses.dataclass

Reply:
Nice, I'll update the code to use these.